lib/app/utils/hl7_validator.rb

```diff
@@ -19,23 +24,32 @@ def validate(resource, fhir_models_klass, profile_url = nil)
 
       result = RestClient.post "#{@validator_url}/validate", resource.source_contents, params: { profile: profile_url }
       outcome = fhir_models_klass.from_contents(result.body)
-      fatals = issues_by_severity(outcome.issue, 'fatal')
-      errors = issues_by_severity(outcome.issue, 'error')
-      warnings = issues_by_severity(outcome.issue, 'warning')
-      information = issues_by_severity(outcome.issue, 'information')
-      {
-        errors: fatals.concat(errors).reject(&:empty?),
-        warnings: warnings,
-        information: information
-      }
+
+      issues_by_severity(outcome.issue)
     end
 
     private
 
-    def issues_by_severity(issues, severity)
-      issues.reject { |i| i.code == 'code-invalid' }
-        .select { |i| i.severity == severity }
-        .map { |iss| "#{issue_location(iss)}: #{iss&.details&.text}" }
+    def issues_by_severity(issues)
+      errors = []
+      warnings = []
+      information = []
+
+      issues.each do |iss|
+        if iss.severity == 'information' || iss.code == 'code-invalid' || DETAILS_FILTER.any? { |filter| filter.match?(iss&.details&.text) }
+          information << "#{issue_location(iss)}: #{iss&.details&.text}"
+        elsif iss.severity == 'warning'
+          warnings << "#{issue_location(iss)}: #{iss&.details&.text}"
+        else
+          errors << "#{issue_location(iss)}: #{iss&.details&.text}"
```

Comment:
Maybe we should create `"#{issue_location(iss)}: #{iss&.details&.text}"` so we don't have to update it in multiple places if we change it.

Reply:
I think creating a method like `issue_message` would be a good idea.